src/Tools/Source/RunTests/Program.cs

```diff
@@ -123,10 +124,36 @@ internal static int Main(string[] args)
                 return ExitFailure;
             }
 
-            Console.WriteLine($"All tests passed");
+            ConsoleUtil.WriteLine($"All tests passed");
             return ExitSuccess;
         }
 
+        private static void LogProcessResultDetails(ImmutableArray<ProcessResult> processResults)
+        {
+            Logger.Log("### Begin logging executed process details");
```

Comment:
Correct. In the case you want it to appear as output of the program then use `ConsoleUtil`. Use normal judgement for what is appropriate there. Diagnostic output that's really only useful for non-trivial investigation goes straight to `Logger.Log`

Note: the options parsing still just uses Console when printing errors. But that is because there really isn't any point in writing to the log if parsing options fails.